logsmanagement/logsmanagement.c

```diff
@@ -0,0 +1,637 @@
+
+    info("Initializing log source collection for %s", filename);
+
+    struct File_info *p_file_info = callocz(1, sizeof(struct File_info));
+
+    p_file_info->filename = (char *) filename;            
+    p_file_info->file_basename = get_basename(filename); // NOTE: get_basename uses strdupz. freez() if necessary!
+    p_file_info->compression_accel = compression_accel;
+    p_file_info->buff_flush_to_db_interval = buff_flush_to_db_interval;
+    p_file_info->blob_max_size = blob_max_size;
+    p_file_info->log_type = log_type;
+    p_file_info->update_every = update_every;
+    p_file_info->circ_buff = circ_buff_init( buff_flush_to_db_interval + CIRCULAR_BUFF_SPARE_ITEMS,
+                                             circular_buffer_max_size);
+
+    /* Add input */
+    switch(log_type){
+        case GENERIC:
+        case WEB_LOG: {
+            rc = tail_plugin_add_input(p_file_info);
+            if(unlikely(rc)){
+                error("tail_plugin_add_input() error for %s: (%d)\n", filename, rc);
+                goto return_on_error;
+            }
+            break;
+        }
+
+        case FLB_GENERIC:
+        case FLB_WEB_LOG:
+        case FLB_SYSTEMD:
+        case FLB_DOCKER_EV: {
+            rc = flb_add_input(p_file_info);
+            if(unlikely(rc)){
+                error("flb_add_input() error for %s: (%d)\n", filename, rc);
+                // m_assert(!rc, "flb_add_input() failed during monitor_log_file_init()");
+                goto return_on_error;
+            }
+
+            p_file_info->flb_tmp_buff_cpy_timer.data = p_file_info;
+            if(unlikely(0 != uv_mutex_init(&p_file_info->flb_tmp_buff_mut))) fatal("uv_mutex_init() failed");
+            uv_timer_init(main_loop, &p_file_info->flb_tmp_buff_cpy_timer);
+            uv_timer_start(&p_file_info->flb_tmp_buff_cpy_timer,
+                        (uv_timer_cb)flb_tmp_buff_cpy_timer_cb, 0, p_file_info->update_every * MS_IN_SEC);            
+            break;
+        }
+
+        default: 
+            goto return_on_error;
+    }
+
+    /* Allocate p_file_info->parser_metrics memory */
+    p_file_info->parser_metrics = callocz(1, sizeof(Log_parser_metrics_t));
+    switch(log_type){
+        case WEB_LOG: 
+        case FLB_WEB_LOG:{
+            p_file_info->parser_metrics->web_log = callocz(1, sizeof(Web_log_metrics_t));
+            break;
+        }
+        case FLB_SYSTEMD: {
+            p_file_info->parser_metrics->systemd = callocz(1, sizeof(Systemd_metrics_t));
+            break;
+        }
+        case FLB_DOCKER_EV: {
+            p_file_info->parser_metrics->docker_ev = callocz(1, sizeof(Docker_ev_metrics_t));
+        }
+        default:
+            break;
+    }
+
+    /* Initialise and create parser thread notifier condition variable and mutex */
+    rc = uv_mutex_init(&p_file_info->notify_parser_thread_mut);
+    if(unlikely(rc)) fatal("Failed to initialise notify_parser_thread_mut for %s\n", p_file_info->filename);
```

Comment:
The function `fatal` shutdown the whole `netdata`. I think we could stop only this collector when this event is not   successful .

Reply:
I've replaced `fatal` errors in various places with termination of the logs management plugin. However, I've kept a few (for example if it cannot create basic structures, such as a mutex or if a memory allocation fails), because I think these are special cases where we _should_ fail gracefully.